Zend/zend_builtin_functions.stub.php

```diff
@@ -69,13 +68,11 @@ function trigger_error(string $message, int $error_type = E_USER_NOTICE): bool {
 /** @alias trigger_error */
 function user_error(string $message, int $error_type = E_USER_NOTICE): bool {}
 
-/** @return mixed */
-function set_error_handler($error_handler, int $error_types = E_ALL) {}
+function set_error_handler($error_handler, int $error_types = E_ALL): ?callable {}
```

Comment:
I'm a bit uncomfortable with these ... how sure are we the return value is still callable within the context where set_error_handler was called? I don't think that's strictly guaranteed.

Reply:
OK, I got the point, I'm not very sure. What about using the `string|array|null` return type then?

Reply:
Assuming you mean `string|array|object|null` here, that seems fine, though somewhat weird.

Reply:
OK, let's just document this in PHPDoc. We can change this later if we want to.